### ISSUE TITLE
Support github style checkboxes

### ISSUE THREAD

**Issue by eric-burel** [slidevjs/slidev#640]
**Is your feature request related to a problem? Please describe.**

I use checkboxes to indicate when something as to be done, when writing slides for courses/practical applications.

According to GitHub, this belongs to their "tasklist" extension: https://github.github.com/gfm/

**Describe the solution you'd like**
This syntax:
```md
- [ ] My text
- [X] Checked 
```
Should translate to:
```html
<li><input type="checkbox" /> My text </li>
<li><input type="checkbox" checked/> My text </li>
```

and render like this (in the web version but also when exported in PDF/PNG):
- [ ] My text
- [X] Checked

I am not sure about the bullet point though, however GitHub works this way currently (just using [ ] without a dash does nothing).

**Describe alternatives you've considered**

- Adding `- <input type="checkbox" />` manually. Not sure how it will behave after export. It doesn't render very well however, because you need a bullet point to have correct line jumps, and the text won't support inline code anymore

- Using a square emoji also works https://emojipedia.org/search/?q=square but is not interactive

**Comment by twitwi** [comment]
maybe https://github.com/revin/markdown-it-task-lists is worth a try (as slidev uses markdown-it)

**Comment by eric-burel** [comment]
@twitwi Is there a doc on how to install such plugins? I find doc to customize almost everything but not the parsing itself, a plugin would be perfect

**Comment by yjl9903** [comment]
See https://sli.dev/custom/config-vite.html#configure-internal-plugins

I think this will work

```ts
import { defineConfig } from 'vite'
import taskLists from 'markdown-it-task-lists'

export default defineConfig({
  slidev: {
    vue: {
      /* vue options */
    },
    markdown: {
      /* markdown-it options */
      markdownItSetup(md) {
        /* custom markdown-it plugins */
        md.use(taskLists)
      },
    },
    /* options for other plugins */
  },
})
```

**Comment by antfu** [comment]
I think it makes sense to have the plugin built in. PR welcome if anyone wants to work on that!

**Comment by antfu** [comment]
We are now using https://github.com/hedgedoc/markdown-it-better-task-lists and shipped in v0.34.3
